gyp/gyp_main.py

```diff
@@ -6,10 +6,38 @@
 
 import os
 import sys
+import subprocess
+
+# Below IsCygwin() function copied from pylib/gyp/common.py
+def IsCygwin():        
+  try:
+    out = subprocess.Popen("uname",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT)
+    stdout,stderr = out.communicate()
+    return "CYGWIN" in stdout
```

Comment:
In Python 3, __stdout__ will be __bytes__ so please pick one of two solutions:
* __return "CYGWIN" in str(stdout)__  # or
* return b"CYGWIN" in stdout